pkg/cloudprovider/backup_service.go

```diff
@@ -155,16 +177,25 @@ func (br *backupService) GetBackup(bucket, name string) (*api.Backup, error) {
 	return backup, nil
 }
 
-func (br *backupService) DeleteBackupFile(bucket, backupName string) error {
-	key := fmt.Sprintf(backupFileFormatString, backupName, backupName)
-	glog.V(4).Infof("Trying to delete bucket=%s, key=%s", bucket, key)
-	return br.objectStorage.DeleteObject(bucket, key)
+func (br *backupService) DeleteBackupDir(bucket, backupName string) error {
```

Comment:
We want to delete metadata file last here (and only if all other deletes succeed), right?

Reply:
I've waffled on this, trying to think about what happens in different scenarios. If we manage to delete the data tarball but fail to delete a logfile, we're left with a logfile and the metadata file. You can't restore, but you could view the logs (not super useful). I'm wondering if it's worth the effort to special case the metadata file?

Reply:
The thinking around deleting the metadata file last was to not orphan any Ark-created object-storage files (or snapshots). As it is now, we could theoretically fail to delete both the tarball and log file, but delete the metadata file, leaving garbage around with no refs to it. I don't think it's a huge deal because the likelihood of this scenario is pretty small IMHO.

Reply:
Maybe for GC we also need to call ListCommonPrefixes and try to delete anything in object storage where the metadata file is missing?

Reply:
let's add an issue and keep moving.

Reply:
Filed https://github.com/heptio/ark/issues/77